### PR TITLE
Cleanup and Plone5 compatiblity

### PR DESCRIPTION
**Pull request by tomgross**

### CODE REVIEW

File: buildout.cfg [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+[versions]
+# Don't use a released version of redomino.tokenrole
+redomino.tokenrole =
+
+setuptools = 20.7.0
+zc.buildout = 2.5.1
+zc.recipe.egg = 2.0.3
+flake8 = 2.5.4
+i18ndude = 4.0.1
+robotframework = 3.0
```
Comment (hvelarde) [inline buildout.cfg]: extend from https://raw.github.com/plone/plone.app.robotframework/master/versions.cfg instead

File: travis.cfg [hunk 2]
```diff
@@ -0,0 +1,12 @@
+[buildout]
+extends =
+    https://raw.githubusercontent.com/collective/buildout.plonetest/master/travis-5.x.cfg
```
Comment (hvelarde) [inline travis.cfg]: why breaking compatibility with Plone 4 explicitly?

**Comment by hvelarde** [comment]
I think is better to keep package compatible with bot Plone 4 and Plone versions when possible

File: travis.cfg (hunk 2 above)
Comment (tomgross) [inline travis.cfg]: This was copied 1:1 from bobtemplates.plone which is considered best practice.
Breaking Plone 4 compatibility is no intention. This is WIP.

Comment (hvelarde) [inline travis.cfg]: seriously? I'll need to take a look at that package to fix it; I have not used a travis.cfg file in eons: that's not best practice anymore; IMO, best practice is to use the container infrastructure on Travis CI.

File: .travis.yml [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+  - sed -ie "s#travis-5.x.cfg#travis-$PLONE_VERSION.x.cfg#" travis.cfg
+  - sed -ie "s#5.0-latest#$PLONE_VERSION-latest#" buildout.cfg
+  - mkdir -p $HOME/buildout-cache/{eggs,downloads}
+  - mkdir $HOME/.buildout
+  - echo "[buildout]" > $HOME/.buildout/default.cfg
+  - echo "download-cache = $HOME/buildout-cache/downloads" >> $HOME/.buildout/default.cfg
+  - echo "eggs-directory = $HOME/buildout-cache/eggs" >> $HOME/.buildout/default.cfg
+install:
+  - python bootstrap-buildout.py -c travis.cfg
+  - bin/buildout -Nc travis.cfg
```
Comment (hvelarde) [inline .travis.yml]: is good practice to include `bin/buildout annotate` here for debugging purposes.

File: .coveragerc [hunk 4]
```diff
@@ -0,0 +1,6 @@
+[report]
```
Comment (hvelarde) [inline .coveragerc]: this is what I normally include:

```
omit =
    */bin/test
    */eggs/*
    */lib/python*
    */parts/*
    */usr/*
```

I don't think is a good idea to omit tests or upgrade steps code.

File: .travis.yml [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+sudo: false
+cache:
+  pip: true
+  directories:
+    - $HOME/buildout-cache
+python:
+  - 2.7
+env:
+  - PLONE_VERSION=4.3
+  - PLONE_VERSION=5.0
```
Comment (hvelarde) [inline .travis.yml]: Plone 5.1 is already out there, you can also added it.

File: setup.py [hunk 6]
```diff
@@ -1,44 +1,60 @@
...
+        'setuptools',
+        'z3c.jbot',
+    ],
+    extras_require={
+        'test': [
+            'plone.app.testing',
+            # Plone KGS does not use this version, because it would break
+            # Remove if your package shall be part of coredev.
+            # plone_coredev tests as of 2016-04-01.
+            'plone.testing>=5.0.0',
```
Comment (hvelarde) [inline setup.py]: this pinning seems to be unnecessary

File: setup.py [hunk 7]
```diff
@@ -1,44 +1,60 @@
...
+    author_email='info@redomino.com',
+    url='https://pypi.python.org/pypi/redomino.tokenrole',
+    license='GPL',
+    packages=find_packages(exclude=['ez_setup']),
+    namespace_packages=['redomino'],
+    include_package_data=True,
+    zip_safe=False,
+    install_requires=[
+        'plone.api',
+        'Products.GenericSetup>=1.8.2',
```
Comment (hvelarde) [inline setup.py]: this pinning seems to be unnecessary

**Comment by tomgross** [comment]
@hvelarde If you have input on good practice for Plone addon boilerplate please consider to place it at the https://github.com/plone/bobtemplates.plone project. This is a boilerplate generator for Plone addons and considered best practice. My addidions here are taken almost taken literally from a generated skeleton.

**Pull request merged**